tests/Microsoft.Identity.Test.Integration.net45/Infrastructure/SeleniumWebUI.cs

```diff
@@ -107,40 +107,72 @@ private IWebDriver InitDriverAndGoToUrl(string url)
             using (var driver = InitDriverAndGoToUrl(authorizationUri.OriginalString))
             using (var listener = new SingleMessageTcpListener(redirectUri.Port)) // starts listening
             {
+                Uri authCodeUri = null;
+                var listenForAuthCodeTask = listener.ListenToSingleRequestAndRespondAsync(
+                    (uri) =>
+                    {
+                        Trace.WriteLine("Intercepted an auth code url: " + uri.ToString());
+                        authCodeUri = uri;
 
-                try
-                {
-                    Uri authCodeUri = null;
-                    var listenForAuthCodeTask = listener.ListenToSingleRequestAndRespondAsync(
-                        (uri) =>
-                        {
-                            Trace.WriteLine("Intercepted an auth code url: " + uri.ToString());
-                            authCodeUri = uri;
+                        return GetMessageToShowInBroswerAfterAuth(uri);
+                    },
+                    cancellationToken);
+
+                // Run the TCP listener and the selenium automation in parallel
+                var seleniumAutomationTask = Task.Factory.StartNew(() => _seleniumAutomationLogic(driver));
 
-                            return GetMessageToShowInBroswerAfterAuth(uri);
-                        },
-                        cancellationToken);
+                await Task.WhenAny(seleniumAutomationTask, listenForAuthCodeTask)
```

Comment:
I changed this logic from Task.WhenAll to Task.WhenAny. WhenAll actually waits for both tasks to finish, even if one fails fast. So any Selenium failure would still wait for the TCP listener to timeout, which takes 1 min. I assume this is causing VSTS to fail. 

When using WhenAny, the await will not throw! So we need to do manual processing of exceptions.